### PR TITLE
Ssl migration

### PR DESCRIPTION
**Pull request by AvihayMReblaze**

### CODE REVIEW

File: src/views/SslList.vue [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+        </li>
+      </ul>
+    </div>
+    <div
+      class="content document-list-wrapper"
+      v-show="!loadingDocCounter && selectedBranch"
+    >
+      <div class="card">
+        <div class="card-content">
+          <div class="content">
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:37]: We can remove this internal card

File: src/views/SslList.vue [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+        if ((this.$route.name as string).includes('SSL/list') && val && val !== oldVal) {
+          this.populateLoadBalancers()
+          this.loadCertificates()
+          this.loadSites()
+        }
+      },
+      immediate: true,
+    },
+  },
+  emits: ['generate-shown-changed', 'call-load-certificate', 'delete-shown-changed', 'second-rowbutton-clicked'],
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:243]: We don't need the `emits` here as this component doesn't emit anything, it only listens to other emits
```suggestion
```

File: src/views/SslList.vue [hunk 3]
```diff
@@ -0,0 +1,689 @@
...
+      rowButtonTitle: 'Edit',
+      secondRowButtonIcon: 'fa-trash',
+      secondRowButtonTitle: 'Delete',
+      isNewLoading: false,
+      titles: DatasetsUtils.titles,
+      loadBalancers: null as Balancer[],
+      certificates: null as Certificate[],
+      configs: [],
+      loadingDocCounter: 0,
+      isDownloadLoading: false,
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:261]: Not being used anywhere

File: src/views/SslList.vue [hunk 4]
```diff
@@ -0,0 +1,689 @@
...
+            if (this.sites?.length > 0) {
+              const matchingSite: Site = _.find(this.sites, (site: Site) => {
+                return site.ssl_certificate === item.id
+              })
+              return matchingSite ? matchingSite.server_names.join('\n') : ''
+            } else {
+              return ''
+            }
+          },
+          isSortable: true,
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:317]: Probably shouldn't be sortable, these values are not linear

File: src/views/SslList.vue [hunk 5]
```diff
@@ -0,0 +1,689 @@
...
+                  return certificateLink.link === balancerCertificateLink
+                })
+              })
+            })
+            const matchingLoadBalancersDNS = _.map(matchingLoadBalancers, (balancer:Balancer) => {
+              return `${balancer.name}\n(${balancer.dns_name})`
+            })
+            return matchingLoadBalancersDNS.join('\n')
+          },
+          isSortable: true,
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:356]: Probably shouldn't be sortable, these values are not linear

File: src/views/SslList.vue [hunk 6]
```diff
@@ -0,0 +1,689 @@
...
+          isSearchable: true,
+          classes: 'width-100px white-space-pre ellipsis vertical-scroll',
+        },
+        {
+          title: 'SAN',
+          fieldNames: ['san'],
+          displayFunction: (item) => {
+            return item?.san?.join('\n')
+          },
+          isSortable: true,
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:366]: Probably shouldn't be sortable, these values are not linear

File: src/views/SslList.vue [hunk 7]
```diff
@@ -0,0 +1,689 @@
...
+        {
+          title: '# Of certs',
+          fieldNames: ['cert'],
+          displayFunction: (item) => {
+            let defaultCertificateNumber = 0
+            if (item?.default_certificate) {
+              defaultCertificateNumber = 1
+            }
+            if (item.provider === 'aws') {
+              return `${item?.certificates?.length + defaultCertificateNumber}/25`
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:394]: What's this `/25`?

File: src/views/SslList.vue [hunk 8]
```diff
@@ -0,0 +1,689 @@
...
+            } else if (item.provider === 'gcp') {
+              return `${item?.certificates?.length + defaultCertificateNumber}/15`
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:396]: What's this `/15`?

File: src/views/SslList.vue [hunk 9]
```diff
@@ -0,0 +1,689 @@
...
+          isSortable: true,
+          isSearchable: true,
+          classes: 'width-100px',
+        },
+        {
+          title: 'Cloud Provider',
+          fieldNames: ['provider'],
+          /* displayFunction: (item) => {
+            return item?.toUpperCase()
+          }, */
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:408]: Remove if not used
```suggestion
```

File: src/views/SslList.vue [hunk 10]
```diff
@@ -0,0 +1,689 @@
...
+          fieldNames: ['load_balancer_type'],
+          isSortable: true,
+          isSearchable: true,
+          classes: 'width-100px white-space-pre',
+        },
+      ] as ColumnOptions[]
+    },
+
+    // This is the specific loadBalancer
+    getBalancerByID() {
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:438]: computed value names usually should not start with `get` - `selectedLoadBalancerID` perhaps?

File: src/views/SslList.vue [hunk 11]
```diff
@@ -0,0 +1,689 @@
...
+      const balancer = this.loadBalancers.find((loadBalancer:any) => {
+        return loadBalancer.id === this.rowClickedId
+      })
+      return balancer
+    },
+
+    documentListAPIPath(): string {
+      const apiPrefix = `${this.apiRoot}/${this.apiVersion}`
+      return `${apiPrefix}/reblaze/configs/${this.selectedBranch}/d/ssl/`
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:447]: Wrong path, it's `/certificates/` I think?

File: src/views/SslList.vue [hunk 12]
```diff
@@ -0,0 +1,689 @@
...
+    },
+
+    branchNames(): string[] {
+      return this.configs?.length ? _.sortBy(_.map(this.configs, 'id')) : []
+    },
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:452]: I think this part can be removed

File: src/views/SslList.vue [hunk 13]
```diff
@@ -0,0 +1,689 @@
...
+  },
+  methods: {
+    generateUUID(): string {
+      let dt = new Date().getTime()
+      return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+        const r = (dt + Math.random() * 16) % 16 | 0
+        dt = Math.floor(dt / 16)
+        return (c == 'x' ? r : (r & 0x3 | 0x8)).toString(16)
+      })
+    },
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:468]: Can be removed, if needed we can use the one in utils

File: src/views/SslList.vue [hunk 14]
```diff
@@ -0,0 +1,689 @@
...
+
+    async addNewCertificate() {
+      this.isNewLoading = true
+      this.generateShown = true
+      this.isNewLoading = false
+    },
+
+    async deleteProfile(id:string) {
+      this.clickedRow = id
+      this.isNewLoading = true
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:531]: You're using `this.isNewLoading` for all kind of methods here, probably meant to use `setLoadingDocStatus`?

File: src/views/SslList.vue [hunk 15]
```diff
@@ -0,0 +1,689 @@
...
+      this.certsSearch = ''
+      // this.selectedBalancer = ''
+    },
+  },
+  async created() {
+    await this.branchesStore.list
+  },
+})
+</script>
+<style lang="scss">
```
Comment (Aviv-Galmidi) [inline src/views/SslList.vue:667]:
```suggestion
<style scoped lang="scss">
```

We don't want to accidently affect other pages

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]
Need to clean the code from the weird vscode formatting
`CertificateEditor.vue` can be removed

File: src/views/SslList.vue (hunk 1 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:37]: Done

File: src/views/SslList.vue (hunk 2 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:243]: Done

File: src/views/SslList.vue (hunk 3 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:261]: Need more instructions

File: src/views/SslList.vue (hunk 4 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:317]: On which column?

File: src/views/SslList.vue (hunk 5 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:356]: On which column?

File: src/views/SslList.vue (hunk 6 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:366]: On which column?

File: src/views/SslList.vue (hunk 7 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:394]: As Shai says to me, Let's talk on that

File: src/views/SslList.vue (hunk 8 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:396]: ^^

File: src/views/SslList.vue (hunk 10 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:438]: Done

File: src/views/SslList.vue (hunk 12 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:452]: Done

File: src/views/SslList.vue (hunk 15 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:667]: Done

File: src/views/SslList.vue (hunk 9 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:408]: Done

File: src/views/SslList.vue (hunk 11 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:447]: Let's talk about that

File: src/views/SslList.vue (hunk 13 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:468]: Done

File: src/views/SslList.vue (hunk 14 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:531]: Done

File: src/views/SslList.vue (hunk 3 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:261]: Done with cleaning more variables that is unused

File: src/views/SslList.vue (hunk 4 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:317]: Done

File: src/views/SslList.vue (hunk 5 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:356]: Done

File: src/views/SslList.vue (hunk 11 above)
Comment (AvihayMReblaze) [reply src/views/SslList.vue:447]: Done

**Pull request closed without merging**